examples/binding_affinity_prediction/main.py

```diff
@@ -23,80 +24,166 @@ def run_a_train_epoch(args, epoch, model, data_loader,
     epoch_loss = 0
     for batch_id, batch_data in enumerate(data_loader):
         indices, ligand_mols, protein_mols, bg, labels = batch_data
-        labels, bg = labels.to(args['device']), bg.to(args['device'])
-        prediction = model(bg)
+        labels = labels.to(args['device'])
+        if args['model'] == 'PotentialNet': 
+            bigraph_canonical, knn_graph = bg # unpack stage1_graph, stage2_graph
+            bigraph_canonical = bigraph_canonical.to(args['device'])
+            knn_graph = knn_graph.to(args['device'])
+            prediction = model(bigraph_canonical, knn_graph)
+        elif args['model'] == 'ACNN':
+            bg = bg.to(args['device'])
+            prediction = model(bg)
         loss = loss_criterion(prediction, (labels - args['train_mean']) / args['train_std'])
         epoch_loss += loss.data.item() * len(indices)
         optimizer.zero_grad()
         loss.backward()
         optimizer.step()
         train_meter.update(prediction, labels)
     avg_loss = epoch_loss / len(data_loader.dataset)
+    # if (args['num_epochs'] - epoch) <= 6: # print only the last 5 epochs
     total_scores = {metric: train_meter.compute_metric(metric, 'mean')
                     for metric in args['metrics']}
     msg = 'epoch {:d}/{:d}, training | loss {:.4f}'.format(
         epoch + 1, args['num_epochs'], avg_loss)
     msg = update_msg_from_scores(msg, total_scores)
     print(msg)
+    return total_scores
 
 def run_an_eval_epoch(args, model, data_loader):
     model.eval()
     eval_meter = Meter(args['train_mean'], args['train_std'])
     with torch.no_grad():
         for batch_id, batch_data in enumerate(data_loader):
             indices, ligand_mols, protein_mols, bg, labels = batch_data
-            labels, bg = labels.to(args['device']), bg.to(args['device'])
-            prediction = model(bg)
+            labels = labels.to(args['device'])
+            if args['model'] == 'PotentialNet':
+                bigraph_canonical, knn_graph = bg # unpack
+                bigraph_canonical = bigraph_canonical.to(args['device'])
+                knn_graph = knn_graph.to(args['device'])
+                prediction = model(bigraph_canonical, knn_graph)
+            elif args['model'] == 'ACNN':
+                bg = bg.to(args['device'])
+                prediction = model(bg)
             eval_meter.update(prediction, labels)
     total_scores = {metric: eval_meter.compute_metric(metric, 'mean')
                     for metric in args['metrics']}
     return total_scores
 
 def main(args):
+    torch.multiprocessing.set_sharing_strategy('file_system')
     args['device'] = torch.device("cuda") if torch.cuda.is_available() else torch.device("cpu")
     set_random_seed(args['random_seed'])
 
-    dataset, train_set, test_set = load_dataset(args)
+    _, train_set, val_set, test_set = load_dataset(args) 
```

Comment:
If we only need `train_set`, `val_set`, and `test_set` from `load_dataset`, then we should directly change `load_dataset`.